src/lib/version/px_update_git_header.py

```diff
@@ -40,7 +40,7 @@
 if args.git_tag:
     git_tag = args.git_tag
 else:
-    git_describe_cmd = 'git describe --exclude ext/* --always --tags --dirty'
+    git_describe_cmd = 'git describe --exclude ext/* --match v[0-9]*.[0-9]*.[0-9]* --abbrev=0 --tags'
```

Comment:
```suggestion
    git_describe_cmd = 'git describe --exclude ext/* --match v[0-9]*.[0-9]*.[0-9]* --tags'
```

Reply:
We may want to keep dirty as well, sometimes someone will put together a build locally without everything committed.